firefox-devtools/profiler
# Persist published profile information to a local store

This PR adds the code to store the profile information to a local indexeddb. This is probably easier to look at individual commits.

One way to test this is:
1/ publish an existing profile from the [deploy preview](https://deploy-preview-2472--perf-html.netlify.app/public/24a9fdbdab160a8b11390de381f1c8b334f50f24/marker-chart/?globalTrackOrder=5-0-2-3-4-1&hiddenGlobalTracks=2-3-4&localTrackOrderByPid=3823-0~&range=0.0000_0.2772&thread=0&v=4), and
2/ look at the indexed db in devtools.

Another way is to use the unfinished UI code from #2639:
1. load that [deploy preview](https://deploy-preview-2639--perf-html.netlify.app/public/24a9fdbdab160a8b11390de381f1c8b334f50f24/marker-chart/?globalTrackOrder=5-0-2-3-4-1&hiddenGlobalTracks=2-3-4&localTrackOrderByPid=3823-0~&range=0.0000_0.2772&thread=0&v=4) and publish the profile again. Notably you can test the case of sanitizing a range.
2. look at the "publish" panel again, there should be a `Delete` button that doesn't do anything.
3. Open the [My Profiles page](https://deploy-preview-2639--perf-html.netlify.app/my-profiles/) and you should see the new profile in the list.

Of course you can try this with more profiles :-)

Fixes #2442
